src/NewsService.Business/FindNewsCommand.cs

```diff
@@ -16,37 +31,132 @@ public class FindNewsCommand : IFindNewsCommand
+      {
+        return null;
+      }
+
+      string errorMessage = "Can not get departments. Please try again later.";
+
+      try
+      {
+        Response<IOperationResult<IGetDepartmentsResponse>> response = await _rcGetDepartments
+          .GetResponse<IOperationResult<IGetDepartmentsResponse>>(IGetDepartmentsRequest.CreateObj(departmentIds));
+
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body.Departments;
+        }
+        else
+        {
+          _logger.LogWarning("Errors while getting departments. Reason: {Errors}",
+            string.Join('\n', response.Message.Errors));
+        }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, errorMessage);
+      }
+
+      errors.Add(errorMessage);
+
+      return null;
+    }
 
     public FindNewsCommand(
       INewsRepository repository,
       INewsInfoMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      ILogger<GetNewsCommand> logger,
+      IBaseFindRequestValidator baseFindValidator)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _rcGetDepartments = rcGetDepartments;
+      _rcGetUsers = rcGetUsers;
+      _logger = logger;
+      _baseFindValidator = baseFindValidator;
     }
 
-    public FindResultResponse<NewsInfo> Execute(FindNewsFilter findNewsFilter)
+    public async Task<FindResultResponse<NewsInfo>> Execute(FindNewsFilter findNewsFilter)
     {
       FindResultResponse<NewsInfo> response = new();
 
-      response.Body = _repository
-        .Find(findNewsFilter, out int totalCount)
-        .Select(_mapper.Map)
-        .ToList();
-
-      response.TotalCount = totalCount;
-
-      if (response.Errors.Any())
+      if (!_baseFindValidator.ValidateCustom(findNewsFilter, out List<string> errors))
       {
         _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
         response.Status = OperationResultStatusType.Failed;
+        response.Errors = errors;
         return response;
       }
 
-      response.Status = OperationResultStatusType.FullSuccess;
+      List<DbNews> dbNewsList = _repository.Find(findNewsFilter, out int totalCount);
+
+      List<Guid> departmentsIds = dbNewsList.Where(d => d.DepartmentId.HasValue).Select(d => d.DepartmentId.Value).Distinct().ToList();
+      List<DepartmentData> departments = await GetDepartments(departmentsIds, response.Errors);
+
+      List<Guid> authorsIds = dbNewsList.Select(a => a.AuthorId).Distinct().ToList();
+      List<UserData> authors = await GetAuthors(authorsIds, response.Errors);
+      List<Guid> imagesIds = new();
+
+      imagesIds.AddRange(authors?.Where(u => u.ImageId.HasValue).Select(u => u.ImageId.Value).ToList());
+
```

Comment:
add request to get avatars of users to image service

Reply:
done